### PR TITLE
up-to-date action can be activated by auto-merge-bot

### PR DESCRIPTION
**Pull request by Bullrich**
- triggered up-to-date when auto-merge is enabled
  - Linked actions so that, when auto-merge is enabled, it will trigger up-to-date to update every PR (including this one)
     - Resolves https://github.com/paritytech/up-to-date-action/issues/14
- Updated `up-to-date` action's version
  - This latest change include paritytech/up-to-date-action#17 which stops the action from updating PRs which are already up to date
- replaced github action for github's own action
  - Let's use github's tools when possible as they _should_ be more secure

- [x] Does not require a CHANGELOG entry

### CODE REVIEW

**Review by muharem — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Comment by Bullrich** [comment]
Hi! This PR is still waiting for an approval.

Can a rank 3 or above please look into it?

CC to possible reviewers: 
@kianenigma
@tomaka
@xlc
@arkpar
@ordian
@noot

**Review by ordian — APPROVED** [review]

**Pull request merged**
